gitoxide-core/src/repository/config.rs

```diff
@@ -57,12 +57,12 @@ struct Filter {
 
 impl Filter {
     fn new(input: BString) -> Self {
-        match gix::config::parse::key(<_ as AsRef<BStr>>::as_ref(&input)) {
-            Some(key) => Filter {
-                name: key.section_name.into(),
-                subsection: key.subsection_name.map(ToOwned::to_owned),
+        match ((&input).section_name(), (&input).subsection_name()) {
+            (section, subsection) if !section.is_empty() => Filter {
+                name: section.into(),
+                subsection: subsection.map(ToOwned::to_owned),
```

Comment:
I'm not sure exactly how the filter is used, so I'm not 100% sure this change retains the exact behavior of the filter in all corner cases.